### PR TITLE
USWDS-Site - Add list accessibility tests page

### LINKED ISSUE #2727: USWDS-Site -  CC a11y: publish usa-list checklist items
## Summary
Publish the `usa-list` accessibility test checklist items found in the [Critical Checklist spreadsheet](https://docs.google.com/spreadsheets/d/1fAJaObQHv7aIaeUpH4fTLmauhmugMtNCrbpZe4BJcPY/edit?usp=sharing) (Google Doc :lock:). 

The process for creating a new accessibility test page can be found in the [Creating accessibility checklists - process doc](https://docs.google.com/document/d/14WTCsdzgwpoz_Miztd66RmEGfwwSEUFCVN1DkSQITTE/edit#heading=h.fa3sv1ray42g) (Google Doc :lock:).

### PR DESCRIPTION
**Pull request by RachelCorsino**
# Summary

Added accessibility tests page for the list component.

## Related issue

Closes #2727 

## Preview link

[Preview link](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/rc-publish-list-checklist-items/components/list/accessibility-tests/) - [`/components/list/accessibility-tests`]

## Testing and review
Follow these steps:

1. Confirm that both the main component page and the accessibility tests page have the correct compliance tag at the top.
2. Confirm that the main component page links to the accessibility tests page in the side navigation.
3. Confirm that the main component page links to the accessibility tests page from the accessibility guidance section.
4. Confirm that the main component page shows the accessibility tests summary.
5. Check that accessibility tests page provides the correct counts for each test status type.
6. Confirm the test checklist summaries and data are accurate.
7. Confirm no visual issues.
8. Confirm there is an appropriate changelog entry on both the main component page and the accessibility tests page.

### CODE REVIEW

File: _data/changelogs/component-list.yml
Comment (mejiaj) [inline _data/changelogs/component-list.yml:5]: **chore**: Need to update this date based on final review/merge.

File: _data/changelogs/component-list-accessibility.yml
Comment (mejiaj) [inline _data/changelogs/component-list-accessibility.yml:4]: **chore:** Need to update this date based on final review/merge.

**Review by mejiaj — APPROVED** [review]
Thanks for creating this!

Add a minor comment about updating the date before final review/merge, otherwise LGTM.

File: _data/accessibility-tests/list.yml
Comment (amyleadem) [inline _data/accessibility-tests/list.yml:21]: **issue:** The "retested" date on this item is listed as 05/02/2024, which means it should be tagged as 3.8.0. 

**note:** The process doc was not clear about including "retested" dates here, so I updated the instructions for finding the `version_tested` value. 

```suggestion
    version_tested: 3.8.0
```

File: _data/accessibility-tests/list.yml
Comment (amyleadem) [inline _data/accessibility-tests/list.yml:29]:
```suggestion
    version_tested: 3.8.0
```

File: _data/accessibility-tests/list.yml
Comment (amyleadem) [inline _data/accessibility-tests/list.yml:37]:
```suggestion
    version_tested: 3.8.0
```

File: _data/accessibility-tests/link.yml
Comment (mahoneycm) [inline _data/accessibility-tests/link.yml:42]: **polish ** Looks like some additional lines were added to the link accessibility test page. Would you mind removing these?

```suggestion
```

**Review by mahoneycm — CHANGES_REQUESTED** [review]
Great work on this @RachelCorsino ! It's looking great. Kudos on the changelogs as well! I left a note to update those once this is approved to be merged but you won't have to worry about that right now.

Requesting two small changes to resolve some visual discrepancies.

File: _data/accessibility-tests/list.yml
Comment (amyleadem) [inline _data/accessibility-tests/list.yml:52]: @amycole501 @alex-hull 

**question:** Is this saying that every list must have a header immediately preceding it in order to be accessible? I'm hesitating on this one because the way I am reading it feels surprising to me, and I want to make sure I understand.

File: _data/accessibility-tests/list.yml
Comment (amyleadem) [inline _data/accessibility-tests/list.yml:34]: @amycole501  @alex-hull 

**issue:** In voiceover,  I hear only "bullet" for unordered list items. I recommend we include that as an acceptable readout for this test. Let me know what you think!

**suggestion:**

```suggestion
      When you listen to an unordered list with a screen reader,
      you hear "bullet item name" or "bullet" before each list item.
```

**Review by amyleadem — CHANGES_REQUESTED** [review]
@RachelCorsino Looks really good! Thanks for taking this on. 

I've added some small notes and questions below (Some of them will need to be answered by @alex-hull and @amycole501). Let me know if you have any questions.

File: _data/accessibility-tests/list.yml
Comment (RachelCorsino) [reply _data/accessibility-tests/list.yml]: updated version tested value to 3.8.0

File: _data/accessibility-tests/list.yml
Comment (RachelCorsino) [reply _data/accessibility-tests/list.yml]: updated version tested value to 3.8.0

File: _data/accessibility-tests/list.yml
Comment (RachelCorsino) [reply _data/accessibility-tests/list.yml]: updated version tested value to 3.8.0

File: _data/accessibility-tests/list.yml
Comment (RachelCorsino) [reply _data/accessibility-tests/list.yml]: updated and added 'or "bullet"' to summary details for "screen reader correctly reads unordered lists"

**Comment by RachelCorsino** [comment]
All fixes have been made and PR should be ready to go now!

This PR is not passing all test due to Snyk alert issue, which as been fixed in the following PR: [USWDS-Site: Updates Snyk Alerts #2751](https://github.com/uswds/uswds-site/pull/2751)

File: _data/accessibility-tests/list.yml
Comment (amyleadem) [inline _data/accessibility-tests/list.yml:45]:
```suggestion
    version_tested: 3.8.0
```

File: _data/accessibility-tests/list.yml
Comment (mahoneycm) [reply _data/accessibility-tests/list.yml:52]: Curious if this falls into the category of "general layout" accessibility guidance, and less in the strictly "list" guidance. Considering a list that comes after a paragraph, I wouldn't expect to hear the heading before the list.

**Review by mahoneycm — APPROVED** [review]
Barring the existing accessibility question, this looks good to me!

**Review by amyleadem — CHANGES_REQUESTED** [review]
Thanks for the quick fixes, @RachelCorsino. I found one more reference to 3.6.1 that should be updated. I've added a suggestion in the comment below. 

Other than that, just waiting to hear back from the accessibility team about the [2.4.6 checklist item](https://github.com/uswds/uswds-site/pull/2746#discussion_r1678101059).

Looking good!

File: _data/accessibility-tests/list.yml
Comment (RachelCorsino) [reply _data/accessibility-tests/list.yml]: Resolved in [e1b0c19](https://github.com/uswds/uswds-site/pull/2746/commits/e1b0c192cb02e4647a06ee6629bcb4bf69166d30)

File: _data/accessibility-tests/list.yml
Comment (RachelCorsino) [reply _data/accessibility-tests/list.yml]: Resolved in [f359dbc](https://github.com/uswds/uswds-site/pull/2746/commits/f359dbc93ef16ed7a18a20dd41c5320ebc2bec63)

File: _data/accessibility-tests/list.yml
Comment (RachelCorsino) [reply _data/accessibility-tests/list.yml]: Resolved in [f359dbc](https://github.com/uswds/uswds-site/pull/2746/commits/f359dbc93ef16ed7a18a20dd41c5320ebc2bec63)

File: _data/accessibility-tests/link.yml
Comment (RachelCorsino) [reply _data/accessibility-tests/link.yml]: Resolved in [f359dbc](https://github.com/uswds/uswds-site/pull/2746/commits/f359dbc93ef16ed7a18a20dd41c5320ebc2bec63)

File: _data/accessibility-tests/list.yml
Comment (RachelCorsino) [reply _data/accessibility-tests/list.yml]: Resolved in [f359dbc](https://github.com/uswds/uswds-site/pull/2746/commits/f359dbc93ef16ed7a18a20dd41c5320ebc2bec63)

File: _data/accessibility-tests/list.yml
Comment (RachelCorsino) [reply _data/accessibility-tests/list.yml]: Resolved in [4b363a2](https://github.com/uswds/uswds-site/pull/2746/commits/4b363a26e8f025ce72ba5530c12524542170e504)

**Comment by mahoneycm** [comment]
## List "Accessibility Guidance" section

After re-reviewing, I remembered that we added the "accessibility guidance" section to the Prose component page in order to display the accessibility page callout section.

image

For consistency, it's probably a good idea to add it here too.

I was able to do this by creating a `_components/prose/guidance/accessibility.md` file with just a comment to auto-generate the section. ([Commit](https://github.com/uswds/uswds-site/pull/2613/commits/f5120eb848f05ea8a27af534d00c6f9d7e21fdf7) for reference)
https://github.com/uswds/uswds-site/blob/f5120eb848f05ea8a27af534d00c6f9d7e21fdf7/_components/prose/guidance/accessibility.md?plain=1#L1

**Comment by RachelCorsino** [comment]
Resolved in [25bc6e4](https://github.com/uswds/uswds-site/pull/2746/commits/25bc6e46900fa38903c5e1c0f3d9608aecd4705d)

> ## List "Accessibility Guidance" section
> After re-reviewing, I remembered that we added the "accessibility guidance" section to the Prose component page in order to display the accessibility page callout section.
> 
> image
> 
> For consistency, it's probably a good idea to add it here too.
> 
> I was able to do this by creating a `_components/prose/guidance/accessibility.md` file with just a comment to auto-generate the section. ([Commit](https://github.com/uswds/uswds-site/pull/2613/commits/f5120eb848f05ea8a27af534d00c6f9d7e21fdf7) for reference)
> 
> https://github.com/uswds/uswds-site/blob/f5120eb848f05ea8a27af534d00c6f9d7e21fdf7/_components/prose/guidance/accessibility.md?plain=1#L1

**Review by mahoneycm — APPROVED** [review]
Looks great! Thanks Rachel 🎉

File: _data/accessibility-tests/list.yml
Comment (alex-hull) [reply _data/accessibility-tests/list.yml]: I think making this change makes sense. Just to cover various screen readers.

File: _data/accessibility-tests/list.yml
Comment (alex-hull) [reply _data/accessibility-tests/list.yml:52]: I think that is a good point, Charlie. Wondering if this something we could add to some list pattern guidance, as I think it is at least best practice to ensure that there is some sort of element telling the user what the list is about. Wondering if we could generalize it and say something like, "When listening with a screen reader, appropriate markup is used to inform the user they have entered a list." Based on [this technique](https://www.w3.org/WAI/WCAG22/Techniques/html/H48.html)

Comment (mahoneycm) [reply _data/accessibility-tests/list.yml:52]: I think that makes sense! Looks like this is related to the `1.3.1` criteria which we outlined already. 

Since the [semantic markup](https://dequeuniversity.com/tips/list-markup) for lists does not include any heading tags, would it be appropriate to remove this test?

Comment (alex-hull) [reply _data/accessibility-tests/list.yml:52]: I think it's cool to remove it! I would love in the future pointing to any guidance directly from these accessibility tests (hopes and dreams for the FAR future).

File: _data/changelogs/component-list-accessibility.yml
Comment (mejiaj) [reply _data/changelogs/component-list-accessibility.yml]: Resolving as duplicate comment.

File: _data/changelogs/component-list.yml
Comment (mejiaj) [reply _data/changelogs/component-list.yml]: Resolving as duplicate comment.

File: _data/accessibility-tests/list.yml
Comment (amyleadem) [reply _data/accessibility-tests/list.yml:52]: Thanks for the conversation @mahoneycm and @alex-hull. @RachelCorsino, can you remove this item from the page?

Comment (RachelCorsino) [reply _data/accessibility-tests/list.yml]: Resolved in [96c11ec](https://github.com/uswds/uswds-site/pull/2746/commits/96c11ec7758894a1c077bf301dd0415293d9d6e4)

**Comment by amyleadem** [comment]
> [!Important]
> The build errors should be resolved once we merge in [USWDS-Site: Updates Snyk Alerts](https://github.com/uswds/uswds-site/pull/2751#top) or update uswds-compile to 1.2.0 (not yet released).

**Review by alex-hull — APPROVED** [review]
Looks good! Approved on my end.

**Review by amycole501 — APPROVED** [review]

File: _data/changelogs/component-list-accessibility.yml
Comment (thisisdano) [inline _data/changelogs/component-list-accessibility.yml:4]:
```suggestion
 - date: 2024-08-13
```

File: _data/changelogs/component-list.yml
Comment (thisisdano) [inline _data/changelogs/component-list.yml:5]:
```suggestion
- date: 2024-08-13
```

**Review by thisisdano — COMMENTED** [review]
Updating changelog dates

**Review by thisisdano — APPROVED** [review]

**Pull request merged**
